### ISSUE TITLE
zstandard keeps full file in memory

### ISSUE THREAD

**Issue by MaxNoe** [cta-observatory/pyeventio#167]
```
from eventio import EventIOFile
import os
import psutil



p = psutil.Process(os.getpid())

path = 'corsika_proton_run_00018421_az240-250_zd01-02.eventio.zst'

for i, o in enumerate(EventIOFile(path)):
    if o.header.type == 1202:
        if o.header.id % 100 == 0:
            print(p.memory_info()[0] / 1024**2)

```

Results in:

```
68.47265625
204.08203125
...
5835.42578125
```

5835 MB being the size of the uncompressed file.

This only affects zstandard files, memory using for using an uncompressed file is around 50 MB.

**Comment by MaxNoe** [comment]
I cannot reproduce this reading only the file using zstdard:


```
import zstandard as zstd
dctx = zstd.ZstdDecompressor()
reader = dctx.stream_reader(open(path, 'rb'))

while True:
    chunk = reader.read(100 * 1024**2)
    print(p.memory_info()[0] / 1024**2)

    if not chunk:
        break
```

Only uses 140 MB of RAM at a time.

**Comment by dneise** [comment]
well spotted ... let the hunt begin

**Comment by MaxNoe** [comment]
Turned out, python-zstandard has a memory leaking bug when seeking as opposed to reading, See indygreg/python-zstandard#82.

A hotfix is in #168, but I will leave this issue open until the fix is upstream and we can remove the hotfix.

**Comment by MaxNoe** [comment]
Ok, fix is in zstandard `v0.11.1`, which now on pypi. I removed the monkey patch in #176 and this now raises a warning if zstandard is older.
